frontend/packages/dev-console/src/components/pipelineruns/PipelineRunDetailsPage.tsx

```diff
@@ -19,7 +20,7 @@ const PipelineRunDetailsPage: React.FC<DetailsPageProps> = (props) => (
         component: PipelineRunLogsWithActiveTask,
       },
     ]}
-    badge={<DevPreviewBadge />}
+    badge={getBadgeFromType(PipelineRunModel.badge)}
```

Comment:
Wouldn't it make sense to move this to the base component (`DetailsPage`)? That way consumers don't need to supply a badge, it will be added automatically based on the model.

Note this same comment applies to several places.

Reply:
Each of the pages that use `getBadgeFromType` are custom functionality pages. I don't think they are compatible with moving to `DetailsPage`

Reply:
Misunderstood what you meant - fix applied.